tempesta_fw/http.c

```diff
@@ -3162,6 +3162,16 @@ tfw_http_resp_cache_cb(TfwHttpMsg *msg)
 	TfwHttpReq *req = resp->req;
 
 	TFW_DBG2("%s: req = %p, resp = %p\n", __func__, req, resp);
+	/*
+	 * Response will never be sent: client has disconnected already.
+	 * Don't adjust response, simply destroy it.
+	 */
+	if (unlikely(test_bit(TFW_HTTP_B_REQ_DROP, req->flags))) {
+		TFW_DBG2("%s: resp=[%p] dropped: client disconnected\n",
```

Comment:
Not only about this PR, but it looks like common issue: it seems that we have race condition if
1 thr.: `TFW_HTTP_B_REQ_DROP` set in `tfw_http_conn_cli_drop()`;
2 thr.: `TFW_HTTP_B_REQ_DROP` tested as 1 here (or in `tfw_http_req_evict_dropped()` or in `tfw_http_req_zap_error()`);
2 thr.: `tfw_http_conn_msg_free((TfwHttpMsg *)req)` processed;
1 thr.: the rest of `tfw_http_conn_cli_drop()` is executing (`list_del_init(&req->msg.seq_list)` etc).